TensorFlow Lite label_image fails to build with cmake

### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

v2.17.0-rc0

### Custom code

No

### OS platform and distribution

Linux (Raspberry Pi OS)

### Mobile device

Raspberry Pi 4

### Python version

3.11.2

### Bazel version

_No response_

### GCC/compiler version

12.2.0

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current behavior?

I get the error 
> tensorflow/lite/profiling/proto/profiling_info.pb.h: No such file or directory.


### Standalone code to reproduce the issue

```shell
$ git clone -b v2.17.0-rc0 https://github.com/tensorflow/tensorflow
$ mkdir tflite_build && tflite_build
$ cmake ../tensorflow/tensorflow/lite/
$ cmake --build . -j3
$ cmake --build . -t label_image
In file included from /home/pi/tensorflow/tensorflow/lite/profiling/profile_summarizer.h:28,
                 from /home/pi/tensorflow/tensorflow/lite/profiling/profile_summarizer.cc:16:
/home/pi/tensorflow/tensorflow/lite/profiling/profile_summary_formatter.h:31:10: fatal error: tensorflow/lite/profiling/proto/profiling_info.pb.h: No such file or directory
   31 | #include "tensorflow/lite/profiling/proto/profiling_info.pb.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.

```
```


### Relevant log output

_No response_

The problem is that `tensorflow/lite/profiling/proto/profiling_info.pb.h` is generated under `CMAKE_BINARY_DIR`, but there is no include_directories setting.

Adding the following modification to `tensorflow/lite/examples/label_image/CMakeLists.txt` will resolve the error, 
```bash
$ git diff tensorflow/lite/examples/label_image/CMakeLists.txt
diff --git a/tensorflow/lite/examples/label_image/CMakeLists.txt b/tensorflow/lite/examples/label_image/CMakeLists.txt
index 9874801f34f..5acbae31565 100644
--- a/tensorflow/lite/examples/label_image/CMakeLists.txt
+++ b/tensorflow/lite/examples/label_image/CMakeLists.txt
@@ -61,6 +61,11 @@ if(TFLITE_ENABLE_EXTERNAL_DELEGATE)
           ${TFLITE_SOURCE_DIR}/tools/delegates/external_delegate_provider.cc)
 endif()
 
+include_directories(label_image
+  PUBLIC
+  ${CMAKE_BINARY_DIR}
+)
+
 add_executable(label_image
   ${TFLITE_LABEL_IMAGE_SRCS}
 )
```

but a new error will occur.
```bash
/usr/bin/ld: CMakeFiles/label_image.dir/__/__/profiling/profile_summary_formatter.cc.o: in function `tflite::profiling::ProfileSummaryProtoFormatter::GenerateOpProfileDataFromDetail(tsl::StatsCalculator::Detail const*, tsl::StatsCalculator const*, tflite::profiling::OpProfileData*) const [clone .part.0]':
profile_summary_formatter.cc:(.text+0x9c): undefined reference to `google::protobuf::internal::ArenaStringPtr::Set(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, google::protobuf::Arena*)'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x218): undefined reference to `google::protobuf::internal::ArenaStringPtr::Set(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, google::protobuf::Arena*)'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x2e8): undefined reference to `tflite::profiling::OpProfilingStat* google::protobuf::Arena::CreateMaybeMessage<tflite::profiling::OpProfilingStat>(google::protobuf::Arena*)'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x304): undefined reference to `tflite::profiling::OpProfilingStat* google::protobuf::Arena::CreateMaybeMessage<tflite::profiling::OpProfilingStat>(google::protobuf::Arena*)'
/usr/bin/ld: CMakeFiles/label_image.dir/__/__/profiling/profile_summary_formatter.cc.o: in function `tflite::profiling::ProfileSummaryProtoFormatter::GenerateSubGraphProfilingData(tsl::StatsCalculator const*, int, std::map<unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::less<unsigned int>, std::allocator<std::pair<unsigned int const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > const&, tflite::profiling::SubGraphProfilingData*) const':
profile_summary_formatter.cc:(.text+0x3748): undefined reference to `google::protobuf::internal::ArenaStringPtr::Set(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, google::protobuf::Arena*)'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x384c): undefined reference to `tflite::profiling::OpProfileData* google::protobuf::Arena::CreateMaybeMessage<tflite::profiling::OpProfileData>(google::protobuf::Arena*)'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x3858): undefined reference to `google::protobuf::internal::RepeatedPtrFieldBase::AddOutOfLineHelper(void*)'
/usr/bin/ld: CMakeFiles/label_image.dir/__/__/profiling/profile_summary_formatter.cc.o: in function `tflite::profiling::ProfileSummaryProtoFormatter::GenerateDelegateProfilingData(tsl::StatsCalculator const*, tflite::profiling::DelegateProfilingData*) const':
profile_summary_formatter.cc:(.text+0x3a24): undefined reference to `tflite::profiling::OpProfileData* google::protobuf::Arena::CreateMaybeMessage<tflite::profiling::OpProfileData>(google::protobuf::Arena*)'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x3a30): undefined reference to `google::protobuf::internal::RepeatedPtrFieldBase::AddOutOfLineHelper(void*)'
/usr/bin/ld: CMakeFiles/label_image.dir/__/__/profiling/profile_summary_formatter.cc.o: in function `tflite::profiling::ProfileSummaryProtoFormatter::GetOutputString(std::map<unsigned int, std::unique_ptr<tsl::StatsCalculator, std::default_delete<tsl::StatsCalculator> >, std::less<unsigned int>, std::allocator<std::pair<unsigned int const, std::unique_ptr<tsl::StatsCalculator, std::default_delete<tsl::StatsCalculator> > > > > const&, tsl::StatsCalculator const&, std::map<unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::less<unsigned int>, std::allocator<std::pair<unsigned int const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > const&) const':
profile_summary_formatter.cc:(.text+0x3ab8): undefined reference to `tflite::profiling::ModelProfilingData::ModelProfilingData(google::protobuf::Arena*, bool)'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x3b2c): undefined reference to `tflite::profiling::SubGraphProfilingData* google::protobuf::Arena::CreateMaybeMessage<tflite::profiling::SubGraphProfilingData>(google::protobuf::Arena*)'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x3b38): undefined reference to `google::protobuf::internal::RepeatedPtrFieldBase::AddOutOfLineHelper(void*)'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x3b8c): undefined reference to `google::protobuf::MessageLite::SerializeAsString[abi:cxx11]() const'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x3b94): undefined reference to `tflite::profiling::ModelProfilingData::~ModelProfilingData()'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x3bbc): undefined reference to `tflite::profiling::DelegateProfilingData* google::protobuf::Arena::CreateMaybeMessage<tflite::profiling::DelegateProfilingData>(google::protobuf::Arena*)'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x3bc8): undefined reference to `google::protobuf::internal::RepeatedPtrFieldBase::AddOutOfLineHelper(void*)'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x3bdc): undefined reference to `tflite::profiling::ModelProfilingData::~ModelProfilingData()'
/usr/bin/ld: CMakeFiles/label_image.dir/__/__/profiling/profile_summary_formatter.cc.o: in function `tflite::profiling::ProfileSummaryProtoFormatter::HandleOutput(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) const':
profile_summary_formatter.cc:(.text+0x3cf4): undefined reference to `tflite::profiling::BenchmarkProfilingData::BenchmarkProfilingData(google::protobuf::Arena*, bool)'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x3d18): undefined reference to `google::protobuf::MessageLite::SerializeToOstream(std::ostream*) const'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x3d28): undefined reference to `tflite::profiling::BenchmarkProfilingData::~BenchmarkProfilingData()'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x3dc0): undefined reference to `google::protobuf::MessageLite::ParseFromString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x3de4): undefined reference to `google::protobuf::MessageLite::ParseFromString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x3f28): undefined reference to `google::protobuf::Message::DebugString[abi:cxx11]() const'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x3f7c): undefined reference to `tflite::profiling::ModelProfilingData* google::protobuf::Arena::CreateMaybeMessage<tflite::profiling::ModelProfilingData>(google::protobuf::Arena*)'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x3f94): undefined reference to `tflite::profiling::ModelProfilingData* google::protobuf::Arena::CreateMaybeMessage<tflite::profiling::ModelProfilingData>(google::protobuf::Arena*)'
/usr/bin/ld: profile_summary_formatter.cc:(.text+0x3fc4): undefined reference to `tflite::profiling::BenchmarkProfilingData::~BenchmarkProfilingData()'
collect2: error: ld returned 1 exit status
gmake[3]: *** [examples/label_image/CMakeFiles/label_image.dir/build.make:409: examples/label_image/label_image] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:7751: examples/label_image/CMakeFiles/label_image.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:7758: examples/label_image/CMakeFiles/label_image.dir/rule] Error 2
gmake: *** [Makefile:2600: label_image] Error 2

Hi @pkgoogle , 

I replicate the issue and i got the same error , can you please take a look ?

`Scanning dependencies of target label_image
[100%] Building CXX object examples/label_image/CMakeFiles/label_image.dir/bitmap_helpers.cc.o
[100%] Building CXX object examples/label_image/CMakeFiles/label_image.dir/label_image.cc.o
[100%] Building CXX object examples/label_image/CMakeFiles/label_image.dir/home/sawantkumar/tensorflow/third_party/xla/xla/tsl/util/stats_calculator.cc.o
[100%] Building CXX object examples/label_image/CMakeFiles/label_image.dir/__/__/profiling/memory_info.cc.o
[100%] Building CXX object examples/label_image/CMakeFiles/label_image.dir/__/__/profiling/profile_summarizer.cc.o
In file included from /home/sawantkumar/tensorflow/tensorflow/lite/profiling/profile_summarizer.h:28,
                 from /home/sawantkumar/tensorflow/tensorflow/lite/profiling/profile_summarizer.cc:16:
/home/sawantkumar/tensorflow/tensorflow/lite/profiling/profile_summary_formatter.h:31:10: fatal error: tensorflow/lite/profiling/proto/profiling_info.pb.h: No such file or directory
   31 | #include "tensorflow/lite/profiling/proto/profiling_info.pb.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[3]: *** [examples/label_image/CMakeFiles/label_image.dir/build.make:115: examples/label_image/CMakeFiles/label_image.dir/__/__/profiling/profile_summarizer.cc.o] Error 1
make[2]: *** [CMakeFiles/Makefile2:8102: examples/label_image/CMakeFiles/label_image.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:8109: examples/label_image/CMakeFiles/label_image.dir/rule] Error 2
make: *** [Makefile:2595: label_image] Error 2`

--- a/tensorflow/lite/examples/label_image/CMakeLists.txt
+++ b/tensorflow/lite/examples/label_image/CMakeLists.txt
@@ -61,6 +61,11 @@ if(TFLITE_ENABLE_EXTERNAL_DELEGATE)
           ${TFLITE_SOURCE_DIR}/tools/delegates/external_delegate_provider.cc)
 endif()
 
+include_directories(label_image
+  PUBLIC
+  ${CMAKE_BINARY_DIR}
+)
+
 add_executable(label_image
   ${TFLITE_LABEL_IMAGE_SRCS}
 )
@@ -78,4 +83,6 @@ target_compile_options(label_image
 )
 target_link_libraries(label_image
   tensorflow-lite
+  profiling_info_proto
+  protobuf
 )

